index.qmd

```diff
@@ -155,121 +159,413 @@ def create_alluvial_links_for_dict(df, category_column, date_column, date_freq='
+    max_year = eeud['Period'].max()
+    data_last_year = eeud[eeud['Period'].isin([max_year])]
+    summed_data = data_last_year.groupby(category).agg({'TJ': 'sum'})
+    max_use = summed_data['TJ'].idxmax()
+
+    return ui.value_box(
+        max_use if len(max_use) <=32 else max_use[:32]+"...",
+        f"{round(summed_data['TJ'].max()/10)*10:,} TJ",
+        "Largest Use",
+        showcase=ui.HTML('''
+            <svg xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi bi-bar-chart-line" viewBox="0 0 16 16">
+                <path d="M11 2a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v12h.5a.5.5 0 0 1 0 1H.5a.5.5 0 0 1 0-1H1v-3a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v3h1V7a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v7h1zm1 12h2V2h-2zm-3 0V7H7v7zm-5 0v-3H2v3z"/>
+            </svg>
+        '''),
+        theme="secondary",
+        showcase_layout="left center",
+    )
+
+#category = 'Fuel' #title_case(input.highlight_category())
+#dict(
+#    title = "Test",
+#    icon = "star",
+#    color = "secondary",
+#    value = eeud.groupby(category)['TJ'].sum().idxmax()
+#)
 ```
 
 ```{python}
-#| content: valuebox
-#| title: "Title"
-dict(
-    icon = "asterisk",
-    color = "secondary",
-    value = "Value"
-)
+
+@render.ui
+def extra_valuebox():
+    return ui.value_box(
+        "Title",
+        "Value",
+        "Subtitle",
+        showcase=ui.HTML('''
+            <svg xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi bi-question-lg" viewBox="0 0 16 16">
+                <path fill-rule="evenodd" d="M4.475 5.458c-.284 0-.514-.237-.47-.517C4.28 3.24 5.576 2 7.825 2c2.25 0 3.767 1.36 3.767 3.215 0 1.344-.665 2.288-1.79 2.973-1.1.659-1.414 1.118-1.414 2.01v.03a.5.5 0 0 1-.5.5h-.77a.5.5 0 0 1-.5-.495l-.003-.2c-.043-1.221.477-2.001 1.645-2.712 1.03-.632 1.397-1.135 1.397-2.028 0-.979-.758-1.698-1.926-1.698-1.009 0-1.71.529-1.938 1.402-.066.254-.278.461-.54.461h-.777ZM7.496 14c.622 0 1.095-.474 1.095-1.09 0-.618-.473-1.092-1.095-1.092-.606 0-1.087.474-1.087 1.091S6.89 14 7.496 14"/>
+            </svg>
+        '''),
+        theme="secondary",
+        showcase_layout="left center",
+    )
 ```
 
+#### Row {height=50%}
 
-### Column {width=40%}
+```{python}
+@render.ui
+def largest_increase():
+    category = title_case(input.highlight_category())
+    max_year = eeud['Period'].max()
+    data_last_two_years = eeud[eeud['Period'].isin([max_year, max_year - 1])]
+    summed_data = data_last_two_years.groupby([category, 'Period']).agg({'TJ': 'sum'}).unstack()
+    summed_data['growth'] = summed_data[('TJ', max_year)] - summed_data[('TJ', max_year - 1)]
+    max_growth = summed_data['growth'].idxmax()
+
+    return ui.value_box(
+        max_growth if len(max_growth) <=32 else max_growth[:32]+"...",
+        f"{round(summed_data['growth'].max()/10)*10:,} TJ",
+        "Largest Increase",
+        showcase=ui.HTML('''
+            <svg xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi bi-graph-up-arrow" viewBox="0 0 16 16">
```

Comment:
Are these the icons for the overview page? It's not that clear from the code. Would it be better to store these outside the code maybe? Like could be in another config file, or an imported python file, or just as SVG files.